src/renderer/grid_renderer.rs

```diff
@@ -227,6 +237,52 @@ impl GridRenderer {
         drawn
     }
 
+    /// Returns the background opacity based on the provided style and whether the style is the default background.
+    ///
+    /// There are two kinds of styles with fallback logic for background opacity:
+    /// - Default background:
+    ///     1. If style blend is greater than 0, use the blend value.
+    ///     2. Otherwise, use the opacity setting.
+    /// - Non-default background:
+    ///     1. If text_background_opacity is provided, use it.
+    ///     2. If style blend is greater than 0, use the blend value.
+    ///     3. Otherwise, use opaque.
+    ///
+    /// To achieve visual consistency, the `text_background_opacity` setting overrides the style blend value.
+    /// Otherwise, the background won't be visible with transparency enabled when pumblend/winblend is set to 100.
+    /// Since all background blends are set to the winblend/pumblend value for floating windows when multi-grid is enabled.
+    ///
+    /// # Returns
+    ///
+    /// A `f32` value representing the background opacity.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// let style = Style { blend: 50 };
+    /// let is_default_background = true;
+    /// let opacity = get_background_opacity(style, is_default_background);
+    /// assert_eq!(opacity, 0.5);
+    /// ```
+    fn get_background_opacity(style: &Style, is_default_background: bool) -> f32 {
+        fn compute_blend_opacity(blend: u8, default: f32) -> f32 {
+            if blend > 0 {
+                (100 - blend) as f32 / 100.0
+            } else {
+                default
+            }
+        }
+
+        let settings = SETTINGS.get::<WindowSettings>();
+        if is_default_background {
+            compute_blend_opacity(style.blend, settings.transparency)
+        } else {
+            settings
+                .text_background_opacity
+                .unwrap_or_else(|| compute_blend_opacity(style.blend, 1.0))
```

Comment:
Sorry for the late response.

In this implementation, I wanted to avoid computing the style blend with text_background_opacity, mainly to prevent an invisible background when setting winblend/pumblend to a value close to 100 (with multi-grid).

Eventually, a value of 1.0 could be used to maintain the default usage of the style blend, but then a value of 0.99 should be used to achieve opaque text background.

Additionally, a new variable defining the alpha composition used in compute_blend_opacity can be introduced to be more explicit and maybe more user friendly.

What do you think will be the best approach?

> [!NOTE]
> I have currently pushed the first approach to the branch, setting text_background_opacity to 1 as the default using style blend